### PR TITLE
feat(nebula_ros): support `launch_hw` parameter in console again to make replaying rosbags easier

### PR DESCRIPTION
**Pull request by mojomex**
## PR Type

- Improvement

## Description

Add the `launch_hw` argument to all vendors' launch files explicitly (instead of having ROS 2 Launch read it from the `config_file`) to enable the following workflow:

With the same config:
1. Connect to a real sensor and record data to a rosbag
2. Later, replay that rosbag with Nebula in offline mode (`launch_hw:=false`)

## Review Procedure

Try and launch each vendor, once with `launch_hw` being set to `true`, and once to `false`.
Nebula should try to connect to the sensor in one case, and print a message saying it's listening to recorded packets in the other.

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR**: Required to pass before the merge.

### CODE REVIEW

File: nebula_ros/launch/continental_launch_all_hw.xml [hunk 1]
```diff
@@ -11,6 +15,7 @@
 
         <node pkg="nebula_ros" exec="continental_ars548_ros_wrapper_node" name="nebula_continental_ars548" output="screen">
             <param from="$(var config_file)" allow_substs="true"/>
+            <param name="launch_hw" value="$(var launch_hw)"></param>
```
Comment (knzo25) [inline nebula_ros/launch/continental_launch_all_hw.xml:18]:
```suggestion
            <param name="launch_hw" value="$(var launch_hw)"/>
```

File: nebula_ros/launch/continental_launch_all_hw.xml [hunk 2]
```diff
@@ -21,7 +26,8 @@
     <group if="$(eval &quot;'$(var sensor_model)' == 'SRR520' &quot;)" >
 
         <node pkg="nebula_ros" exec="continental_srr520_ros_wrapper_node" name="nebula_continental_srr520" output="screen">
-	    <param from="$(var config_file)" allow_substs="true"/>
+	        <param from="$(var config_file)" allow_substs="true"/>
+            <param name="launch_hw" value="$(var launch_hw)"></param>
```
Comment (knzo25) [inline nebula_ros/launch/continental_launch_all_hw.xml:30]:
```suggestion
            <param name="launch_hw" value="$(var launch_hw)"/>
```

File: nebula_ros/launch/hesai_launch_all_hw.xml [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
   <arg name="sensor_model" description="Pandar64|Pandar40P|PandarXT32|PandarXT32M|PandarAT128|PandarQT64|PandarQT128|Pandar128E4X"/>
+  <arg name="launch_hw" default="true" description="Whether to connect to a real sensor (true) or to accept packet messages (false).">
+    <choice value="true" />
+    <choice value="false" />
+  </arg>
   <arg name="config_file" default="$(find-pkg-share nebula_ros)/config/lidar/hesai/$(var sensor_model).param.yaml"/>
 
   <node pkg="nebula_ros" exec="hesai_ros_wrapper_node" name="hesai_ros_wrapper_node" output="screen">
     <param from="$(var config_file)" allow_substs="true"/>
+    <param name="launch_hw" value="$(var launch_hw)"></param>
```
Comment (knzo25) [inline nebula_ros/launch/hesai_launch_all_hw.xml:12]:
```suggestion
    <param name="launch_hw" value="$(var launch_hw)"/>
```

File: nebula_ros/launch/robosense_launch_all_hw.xml [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
   <arg name="sensor_model" description="Helios|Bpearl"/>
+  <arg name="launch_hw" default="true" description="Whether to connect to a real sensor (true) or to accept packet messages (false).">
+    <choice value="true" />
+    <choice value="false" />
+  </arg>
   <arg name="config_file" default="$(find-pkg-share nebula_ros)/config/lidar/robosense/$(var sensor_model).param.yaml"/>
 
   <node pkg="nebula_ros" exec="robosense_ros_wrapper_node" name="robosense_ros_wrapper_node" output="screen">
     <param from="$(var config_file)" allow_substs="true"/>
+    <param name="launch_hw" value="$(var launch_hw)"></param>
```
Comment (knzo25) [inline nebula_ros/launch/robosense_launch_all_hw.xml:12]:
```suggestion
    <param name="launch_hw" value="$(var launch_hw)"/>
```

File: nebula_ros/launch/velodyne_launch_all_hw.xml [hunk 5]
```diff
@@ -1,10 +1,15 @@
...
   <arg name="sensor_model" description="VLP16|VLP32|VLS128"/>
+  <arg name="launch_hw" default="true" description="Whether to connect to a real sensor (true) or to accept packet messages (false).">
+    <choice value="true" />
+    <choice value="false" />
+  </arg>
   <arg name="config_file" default="$(find-pkg-share nebula_ros)/config/lidar/velodyne/$(var sensor_model).param.yaml"/>
 
   <node pkg="nebula_ros" exec="velodyne_ros_wrapper_node" name="velodyne_ros_wrapper_node" output="screen">
     <param from="$(var config_file)" allow_substs="true"/>
+    <param name="launch_hw" value="$(var launch_hw)"></param>
```
Comment (knzo25) [inline nebula_ros/launch/velodyne_launch_all_hw.xml:12]:
```suggestion
    <param name="launch_hw" value="$(var launch_hw)"/>
```

**Review by knzo25 — CHANGES_REQUESTED** [review]
Left just some change requests to be consistent with the tags

File: nebula_ros/launch/continental_launch_all_hw.xml [hunk 6]
```diff
@@ -24,7 +29,8 @@
     <group if="$(eval &quot;'$(var sensor_model)' == 'SRR520' &quot;)" >
 
         <node pkg="nebula_ros" exec="continental_srr520_ros_wrapper_node" name="nebula_continental_srr520" output="screen">
-	    <param from="$(var config_file)" allow_substs="true"/>
```
Comment (knzo25) [inline nebula_ros/launch/continental_launch_all_hw.xml:27]: There seems to be a missalign in this line :sweat_smile:

Comment (mojomex) [reply nebula_ros/launch/continental_launch_all_hw.xml:27]: Ah, that misaligned line was removed in this PR. The corrected line is line `32` below :raised_hands:

**Review by knzo25 — APPROVED** [review]
LGTM

**Pull request merged**